From a5377478c8bc142710cd1af64eb6874e4855b2e9 Mon Sep 17 00:00:00 2001
From: "Thulasimani,Sivakumar" <sivakumar.thulasimani@intel.com>
Date: Tue, 6 Oct 2015 17:52:45 +0530
Subject: [PATCH 14/28] FOR_UPSTREAM [VPG]: drm/i915: Avoid optimizing lane
 count

Automated tests ran as part of compliance tests expect lane count
to be the same as reported in DPCD, so we should avoid
optimization of lane count during such tests. This patch updates
the lane count to max supported value when link training test
is in progress

v2: perform this for all scenarios
v3: all scenarios results in underrun, so rolled back to original
case of using max lane count only for test scenarios

Change-Id: I0b6010d2207e933fd65026e70467387e48278359
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-9749
Signed-off-by: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
Reviewed-on: https://android.intel.com:443/439687
---
 drivers/gpu/drm/i915/intel_dp.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 126ee92..59ff3fe 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -1093,6 +1093,13 @@ found:
 	if (intel_dp->color_range)
 		pipe_config->limited_color_range = true;
 
+	/*
+	 * compliance tests expect same lane count as reported, so
+	 * avoid any optimization during tests
+	 */
+	if (intel_dp->compliance_test_type == DP_TEST_LINK_TRAINING)
+		lane_count = max_lane_count;
+
 	intel_dp->link_bw = bws[clock];
 	intel_dp->lane_count = lane_count;
 	pipe_config->pipe_bpp = bpp;
-- 
1.9.1

